File: backend/Models/Order.cs
using System.ComponentModel.DataAnnotations;

namespace BuckeyeMarketplace.Models;

public class Order
{
    public int Id { get; set; }

    public string? UserId { get; set; }

    public DateTime OrderDate { get; set; } = DateTime.UtcNow;

    [Required]
    public string ConfirmationNumber { get; set; } = string.Empty;

    [Required]
    public string ShippingAddress { get; set; } = string.Empty;

    public decimal Total { get; set; }

    [Required]
    public string Status { get; set; } = "Placed";

    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
}

File: frontend/src/types/order.ts
export interface OrderItem {
  productId: number;
  productTitle: string;
  unitPrice: number;
  quantity: number;
}

export interface Order {
  id: number;
  userId: string | null;
  orderDate: string;
  confirmationNumber: string;
  shippingAddress: string;
  total: number;
  status: string;
  items: OrderItem[];
}

export interface GuestCartItem {
  productId: number;
  quantity: number;
}

export interface PlaceOrderRequest {
  shippingAddress: string;
  items?: GuestCartItem[];
}
